metals/src/main/scala/scala/meta/internal/metals/debug/DebugProvider.scala

```diff
@@ -407,13 +413,25 @@ class DebugProvider(
 
   private val reportErrors: PartialFunction[Throwable, Unit] = {
     case _ if buildClient.buildHasErrors =>
-      statusBar.addMessage(Messages.DebugErrorsPresent(icons))
-      languageClient.metalsExecuteClientCommand(
-        new ExecuteCommandParams(
-          ClientCommands.FocusDiagnostics.id,
-          Nil.asJava
-        )
-      )
+      val msg =
+        "Cannot run or debug due to existing errors in the workspace. " +
+          "Please fix the errors and retry."
+      clientConfig.statusBarState() match {
+        case On =>
+          statusBar.addMessage(Messages.DebugErrorsPresent(icons, msg))
```

Comment:
Could we actually handle this all in `statusBar`? This way even if someone forgets to properly check statusBar config, it would work. This would most likely deduplicate the code below also.

`FocusDiagnostics` should probably be handled by a different option.

Reply:
> Could we actually handle this all in statusBar? This way even if someone forgets to properly check statusBar config, it would work. This would most likely deduplicate the code below also.

Ah, actually it looks like I can just use this:

https://github.com/scalameta/metals/blob/9ac13b22569f56128527abe424e556725fb62f57/metals/src/main/scala/scala/meta/internal/metals/ConfiguredLanguageClient.scala#L35-L49

> FocusDiagnostics should probably be handled by a different option.

What do you mean?

Reply:
I just meant that if we want to send the focus diagnostics command then we should check if that is handled by the client irrelevant of the status bar.

Reply:
ahhh ok, gotcha yea. Agreed.